js/reports.jsx

```diff
@@ -66,61 +64,174 @@ var App = React.createClass({
-        </form>
-    )
-  }
+    handleSubmit: function(e) {
+        e.preventDefault();
+        var runSeriesID = this.refs.searchbox.getDOMNode().value;
+        /* 1. Call the parent's search handler */
+        this.props.getResultData(runSeriesID);
+    },
+    render: function() {
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <input type="text" id="searchInput" ref="searchbox" />
+                <button type="submit" id="searchSubmit">Search</button>
+            </form>
+        )
+    }
+});
+
+var GetProductList = React.createClass({
+    printProducts: function(){
+        if (this.props.productNameData != null) {
+            return this.props.productNameData.objects.map(function (productInfo){
+                return <option id={productInfo.id} value={productInfo.name} label={productInfo.name}>{productInfo.name}</option>;
+            }, this);
+        }
+    },
+
+    componentDidUpdate: function(){
+        this.props.updateCurrentProductName(this.refs.productList.getDOMNode().value)
+    },
+
+    render: function(){
+        var options = this.printProducts();
+        return(
+            <select id="productList" ref="productList" onChange={this.props.productNameOnChange}>
+                {options}
+            </select>
+        )
+    }
+});
+
+var GetProductVersionList = React.createClass({
+    printProducts: function(){
+        if (this.props.productVersionData != null) {
+            return this.props.productVersionData.objects.map(function (productInfo){
+                return <option id={productInfo.id} value={productInfo.version} label={productInfo.version}>{productInfo.version}</option>;
+            }, this);
+        }
+    },
+
+    componentDidUpdate: function(){
+      this.props.updateCurrentProductVersion(this.refs.productVersionList.getDOMNode().value)
+    },
+
+    render: function(){
+        var options = this.printProducts();
+        return(
+            <select id="productVersionList" ref="productVersionList" onChange={this.props.productVersionOnChange}>
+                {options}
+            </select>
+        )
+    }
 });
 
 var HistoryReport = React.createClass({
-  getInitialState: function(){
-    return ({ data: null
-    })
-  },
-
-  getData: function(id) {
-    var url = "http://127.0.0.1:8000/api/v1/resultview/?format=json&limit=0&runcaseversion__run__series=" + id
-    $.ajax({
-      url: url,
-
-      success: function(data) {
-        this.setState({data: data});
-      }.bind(this),
-
-      error: function(xhr, status, err) {
-        //this.setState(this.notFound)
-        console.error(xhr, status, err.toString());
-      }.bind(this)
-    });
+    getInitialState: function(){
+        return ({ resultData: null,
+            currentProductName: null,
+            currentProductVersion: null,
+            productNameData: null,
+            productVersionData: null,
+            initProductVersionFlag: null
```

Comment:
Can we have a more semantic flag instead of `null`/`1`?  Perhaps a `isInit...=true/false` or change the content into something like `"init"/"ready"`